core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java

```diff
@@ -196,6 +200,12 @@ void doMarshall(SdkPojo pojo) {
         }
     }
 
+    private void updateContentLengthHeader(int contentLength) {
+        if (!request.firstMatchingHeader(CONTENT_LENGTH).isPresent()) {
+            request.putHeader(CONTENT_LENGTH, Integer.toString(contentLength));
+        }
```

Comment:
When will this condition be needed? Why not just set it?

Reply:
Thank you for catching that oversight. I assumed that "MergeCustomHeadersStage" would override the headers before this. However, this stage comes after Marshalling is done. I removed this check.